From dfef3032725b1455dba46456aa21366b495e289f Mon Sep 17 00:00:00 2001
From: Al Muthanna Athamina <almuthanna@chromium.org>
Date: Wed, 18 Aug 2021 10:31:10 +0200
Subject: [PATCH] [infra] Add loong64 ports waterfall and opt-in trybot

Bug: v8:12100
Change-Id: I35d71d7649221febcf911010577330e1a264dc34
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3103005
Reviewed-by: Liviu Rau <liviurau@chromium.org>
Commit-Queue: Almothana Athamneh <almuthanna@chromium.org>
Cr-Commit-Position: refs/heads/main@{#76366}
---
 infra/mb/mb_config.pyl     |  9 +++++++++
 infra/testing/builders.pyl | 23 +++++++++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/infra/mb/mb_config.pyl b/infra/mb/mb_config.pyl
index 236dc1e8474..4354d3fee06 100644
--- a/infra/mb/mb_config.pyl
+++ b/infra/mb/mb_config.pyl
@@ -194,6 +194,8 @@
       'V8 Linux - s390x - sim': 'release_simulate_s390x',
       # RISC-V
       'V8 Linux - riscv64 - sim - builder': 'release_simulate_riscv64',
+      # LOONG-V
+      'V8 Linux - loong64 - sim - builder': 'release_simulate_loong64',
     },
     'tryserver.v8': {
       'v8_android_arm_compile_rel': 'release_android_arm',
@@ -237,6 +239,7 @@
       'v8_linux64_asan_rel_ng': 'release_x64_asan_minimal_symbols',
       'v8_linux64_cfi_rel_ng': 'release_x64_cfi',
       'v8_linux64_fuzzilli_ng': 'release_x64_fuzzilli',
+      'v8_linux64_loong64_rel_ng': 'release_simulate_loong64',
       'v8_linux64_msan_rel_ng': 'release_simulate_arm64_msan_minimal_symbols',
       'v8_linux64_riscv64_rel_ng': 'release_simulate_riscv64',
       'v8_linux64_tsan_rel_ng': 'release_x64_tsan_minimal_symbols',
@@ -413,6 +416,8 @@
       'release_bot', 'simulate_arm64', 'msan_no_origins'],
     'release_simulate_arm64_trybot': [
       'release_trybot', 'simulate_arm64'],
+    'release_simulate_loong64': [
+      'release_bot', 'simulate_loong64'],
     'release_simulate_mipsel': [
       'release_bot', 'simulate_mipsel'],
     'release_simulate_mips64el': [
@@ -823,6 +828,10 @@
       'gn_args': 'target_cpu="x64" v8_target_cpu="arm64"',
     },
 
+    'simulate_loong64': {
+      'gn_args': 'target_cpu="x64" v8_target_cpu="loong64"',
+    },
+
     'simulate_mipsel': {
       'gn_args':
           'target_cpu="x86" v8_target_cpu="mipsel" mips_arch_variant="r2"',
diff --git a/infra/testing/builders.pyl b/infra/testing/builders.pyl
index b64085df1ff..15accfbd727 100644
--- a/infra/testing/builders.pyl
+++ b/infra/testing/builders.pyl
@@ -613,6 +613,16 @@
     ],
   },
   ##############################################################################
+  # Linux64 with LOONG-V simulators
+  'v8_linux64_loong64_rel_ng_triggered': {
+    'swarming_dimensions': {
+      'os': 'Ubuntu-18.04',
+    },
+    'tests': [
+      {'name': 'v8testing', 'shards': 3},
+    ],
+  },
+  ##############################################################################
   # Linux64 with RISC-V simulators
   'v8_linux64_riscv64_rel_ng_triggered': {
     'swarming_dimensions': {
@@ -1862,6 +1872,19 @@
       },
     ],
   },
+  'V8 Linux - loong64 - sim': {
+    'swarming_dimensions': {
+      'os': 'Ubuntu-18.04',
+    },
+    'swarming_task_attrs': {
+      'expiration': 14400,
+      'hard_timeout': 3600,
+      'priority': 35,
+    },
+    'tests': [
+      {'name': 'v8testing', 'shards': 3},
+    ],
+  },
   'V8 Linux - mips64el - sim': {
     'swarming_dimensions': {
       'os': 'Ubuntu-18.04',
-- 
2.35.1

